server/api/controllers/user.js

```diff
@@ -242,9 +240,91 @@ export const followUser = async (req, res) => {
       data: followedUser
     });
   } catch (error) {
-    return res.status(500).send({
+    res.status(500).send({
+      status: 'error',
+      message: 'Internal server error occured.'
+    });
+  }
+};
+export const resetPasswordVerification = async (req, res) => {
+  const {
+    email
+  } = req.body;
+
+  try {
+    const foundUser = await User.findOne({ where: { email: { [Op.eq]: email } } });
+
+    if (!foundUser) {
+      return res.status(404).json({
+        status: 'fail',
+        message: 'User not found,Provide correct email address',
+      });
+    }
+    const token = signToken({
+      email,
+    });
+    const {
+      username,
+    } = foundUser;
+    try {
+      await resetPasswordVerificationMail(username, foundUser.email, token);
+    } catch (error) {
+      logger.debug('Email Error::', error);
+    }
+
+    return res.status(200).send({
+      status: 'success',
+      data: {
+        token,
+        email: foundUser.email,
+        user: `A confirmation email has been sent to ${foundUser.email}. Click on the confirmation button to verify the account`,
+        link: `${req.protocol}://${req.headers.host}/api/users/resetPassword/${token}`
+      },
+    });
+  } catch (e) {
+    res.status(500).send({
       status: 'error',
-      message: 'Internal server error occured',
+      message: 'Internal server error occured.'
+    });
+  }
+};
+
+/**
+ * @param {Object} req - request received
+ * @param {Object} res - response object
+ * @returns {Object} response object
+ */
+export const resetPassword = async (req, res) => {
+  try {
+    const { token } = req.params;
+    const { email } = verifyToken(token);
+    const { password } = req.body;
+
+
+    // Check if the user exists in the database
+    const foundUser = await User.findOne({ where: { email: { [Op.eq]: email } } });
+
+    if (!foundUser) {
+      return res.status(404).json({
+        status: 'fail',
+        message: 'User not found,Provide correct email address',
```

Comment:
@oyedejipeace The only time this check on line `307` will run is if the token is compromised. Not sure if we should have it though. Please, consider using  `res.json` only when u have to send a `json` object explicitly. Please use a `.send` for conformity with what exists in the code base, and because it already adapts the `json` protocol in it.